src/commands/dev/mod.rs

```diff
@@ -43,12 +41,16 @@ pub fn dev(
     let session_id = get_session_id()?;
     let preview_id = get_preview_id(target, user, &server_config, &session_id)?;
 
-    // create a new thread to listen for devtools messages
-    thread::spawn(move || socket::listen(session_id));
-
-    // spawn tokio runtime on the main thread to handle incoming HTTP requests
     let mut runtime = TokioRuntime::new()?;
-    runtime.block_on(serve(server_config, preview_id))?;
+
+    let devtools_listener = socket::listen(session_id);
+    let server = serve(server_config, preview_id);
+
+    let runners = futures::future::join(devtools_listener, server);
+
+    runtime.block_on(async {
+        let (_, _) = runners.await;
```

Comment:
any reason to not 

    let _ =

instead?

Reply:
also, are you sure that ignoring all failures is what you want here?

Reply:
That's a good point - I'm going to add some `.unwrap()` calls to them. Not quite sure if there's a way for me to bubble up those failures through the async fn

Reply:
pub fn block_on<F: Future>(&mut self, future: F) -> F::Output

should return that `Result` back out of `block_on`, I think.

Reply:
very interesting